checker.cs

```diff
@@ -1,34 +1,51 @@
 using System;
 using System.Diagnostics;
 
-class Checker
+internal class Checker
 {
-    static bool vitalsAreOk(float bpm, float spo2, float respRate) {
-        if(bpm < 70 || bpm > 150) {
-            return false;
-        } else if(spo2 < 90) {
-            return false;
-        } else if(respRate < 30 || respRate > 95) {
-            return false;
-        }
-        return true;
-    }
-    static void ExpectTrue(bool expression) {
-        if(!expression) {
-            Console.WriteLine("Expected true, but got false");
-            Environment.Exit(1);
-        }
+    //Convection followed: lower, upper
+    private readonly int[] _spo2Limits  = new int[] {90 , 100};
+    private readonly int[] _respRateLimits = new int[] {30 , 95};
+    private readonly int[] _bpmLimits = new int[] {70 , 150};
+
+    public bool CheckVitals(float bpm, float spo2, float respRate)
+    {
+        return (
+            CheckIfInLimits(_spo2Limits[0], _spo2Limits[1], spo2, "spo2") && 
+            CheckIfInLimits(_respRateLimits[0], _respRateLimits[1], respRate, "RespRate") &&
+            CheckIfInLimits(_bpmLimits[0], _bpmLimits[1], bpm, "BPM") );
+
     }
-    static void ExpectFalse(bool expression) {
-        if(expression) {
-            Console.WriteLine("Expected false, but got true");
-            Environment.Exit(1);
-        }
+
+    private bool CheckIfInLimits(float lowerLimit, float upperLimit, float value, string parameter)
+    {
+        if (!(value > upperLimit) && !(value < lowerLimit)) return true;
+        System.Console.WriteLine("Warning!!!!" + parameter + " value is out of range!" + parameter +" = " + value);
```

Comment:
So it's only a console alert and we cannot plugin other reporters? So if I want to send an SMS alert, I'll need to copy the entire program and alter this line?